examples/plot_how_train_test_and_tune.py

```diff
@@ -0,0 +1,669 @@
+# Let's visualize the second option with a util function.
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+def plot_train_valid_test(ax, windows_dataset, train_subset, val_subset, test_set):
+    """Create a sample plot for training, validation, testing."""
+
+    braindecode_cmap = [
+        "#3A6190",
+        "#683E00",
+        "#2196F3",
+        "#DDF2FF",
+    ]
+    ax.scatter(
+        range(len(windows_dataset)),
+        [3.5] * len(windows_dataset),
+        c=braindecode_cmap[0],
+        marker="_",
+        lw=50,
+    )
+
+    ax.scatter(
+        range(len(train_subset) + len(val_subset) + len(test_set)),
+        [0.5] * len(train_subset) + [0.5] * len(val_subset) + [0.5] * len(test_set),
+        c=[braindecode_cmap[1]] * len(train_subset)
+        + [braindecode_cmap[2]] * len(val_subset)
+        + [braindecode_cmap[3]] * len(test_set),
+        marker="_",
+        lw=50,
+    )
+
+    ax.set(
+        ylim=[-1, 5],
+        yticks=[0.5, 3.5],
+        yticklabels=["Train-Test\nSplit", "Original\nDataset"],
+        xlabel="Number of samples.",
+        title="Train-Validation-Test Split",
+    )
+
+    ax.legend(
+        [
+            Patch(color=braindecode_cmap[0]),
+            Patch(color=braindecode_cmap[1]),
+            Patch(color=braindecode_cmap[2]),
+            Patch(color=braindecode_cmap[3]),
+        ],
+        ["Original set", "Training set", "Validation set", "Testing set"],
+        loc=(1.02, 0.8),
+    )
+
+    return ax
+
+
+fig, ax = plt.subplots(figsize=(12, 5))
+plot_train_valid_test(
+    ax=ax,
+    windows_dataset=windows_dataset,
+    train_subset=train_subset,
+    val_subset=val_subset,
+    test_set=test_set,
+)
+fig.tight_layout()
+
+######################################################################
+# Option 3: k-Fold Cross Validation
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+
+######################################################################
+# As mentioned above, using only one validation split might not be
+# sufficient, as there might be a shift in the data distribution.
+# To compensate for this, one can run a k-fold Cross Validation,
+# where every sample of the training set is in the validation set once.
+# After averaging over the k validation scores afterwards, you get a
+# very reliable estimate of how the model would perform on unseen
+# data (test set).
+#
+# .. note::
+#    This k-Fold Cross Validation can be used without a separate
+#    (holdout) test set. If there is no test set available, e.g. in a
+#    competition, this scheme is highly recommended to get a reliable
+#    estimate of the generalization performance.
+#
+# To implement this, we will make use of sklearn function
+# `cross_val_score <https://scikit-learn.org/stable/modules/generated/
+# sklearn.model_selection.cross_val_score.html>`__ and the `KFold
+# <https://scikit-learn.org/stable/modules/generated/sklearn.model_
+# selection.KFold.html>`__. cross-validator.
```

Comment:
its not called a cross-validator in sklearn but a cross-validation splitter

Reply:
I had the same thought about that one as they call it CV generator or CV splitter in general but cross-validator
In the [KFold doc](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.KFold.html).
The class itself also inherits from `BaseCrossValidator`. 
How should we call it?